source/cipher_openssl.c

```diff
@@ -441,6 +446,10 @@ int aws_cryptosdk_sig_sign_start_keygen(
         goto rethrow;
     }
 
+    // If pub_key is NULL the conversion form is never set, and so it differs between the EC_KEY and EC_GROUP objects.
+    // If this is not a problem this line can be removed, but ec_key_is_valid needs to be changed in the CBMC model.
+    EC_KEY_set_conv_form(keypair, POINT_CONVERSION_COMPRESSED);
```

Comment:
Did this represent a bug?

Reply:
I'm not sure, I need @bdonlan to comment on this. The invariant for EC_KEY currently requires the point conversion form to match the group's point conversion form, but it's not clear if that must always be the case.

There is what I believe is an actual bug below, a NULL-pointer dereference in the cleanup code.